The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8484054430/job/23246309159) [(plain)](https://github.com/rust-lang-ci/rust/commit/ca8923a7036e2682540f090efedfc7b0bb2e224c/checks/23246309159/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [codegen] tests/codegen/vec_pop_push_noop.rs stdout ----

error: verification with 'FileCheck' failed
Build completed unsuccessfully in 0:11:31
Build completed unsuccessfully in 0:11:31
command: "/usr/lib/llvm-17/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/vec_pop_push_noop/vec_pop_push_noop.ll" "/checkout/tests/codegen/vec_pop_push_noop.rs" "--check-prefix=CHECK" "--check-prefix" "NONMSVC" "--allow-unused-prefixes" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/codegen/vec_pop_push_noop.rs:22:12: error: CHECK: expected string not found in input
/checkout/tests/codegen/vec_pop_push_noop.rs:22:12: error: CHECK: expected string not found in input
 // CHECK: call {{.*}}reserve_for_push
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/vec_pop_push_noop/vec_pop_push_noop.ll:287:24: note: scanning from here
define void @push_byte(ptr noalias nocapture noundef align 8 dereferenceable(24) %v) unnamed_addr #0 personality ptr @rust_eh_personality {
                       ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/vec_pop_push_noop/vec_pop_push_noop.ll:320:27: note: possible intended match here
; core::fmt::num::<impl core::fmt::UpperHex for usize>::fmt

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/vec_pop_push_noop/vec_pop_push_noop.ll
Check file: /checkout/tests/codegen/vec_pop_push_noop.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
            .
            .
            .
            .
          187: bb15: ; preds = %bb8 
          188:  store i64 %0, ptr %19, align 8 
          189:  store i64 %1, ptr %20, align 8 
          190:  br label %bb9 
          191:  
          192: bb9: ; preds = %bb16, %bb15, %bb10 
          193:  %storemerge28 = phi i64 [ 1, %bb10 ], [ 0, %bb16 ], [ 1, %bb15 ] 
          194:  store i64 %storemerge28, ptr %_0, align 8 
          195:  ret void 
          196: } 
          198: ; alloc::raw_vec::RawVec<T,A>::grow_one 
          198: ; alloc::raw_vec::RawVec<T,A>::grow_one 
          199: ; Function Attrs: noinline nonlazybind uwtable 
          200: define internal fastcc void @"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$8grow_one17h1f9c44696f70c80fE"(ptr noalias nocapture noundef align 8 dereferenceable(16) %self) unnamed_addr #3 personality ptr @rust_eh_personality { 
          201: start: 
          202:  %_20.i = alloca %"core::option::Option<(core::ptr::non_null::NonNull<u8>, core::alloc::layout::Layout)>", align 8 
          203:  %self3.i = alloca %"core::result::Result<core::ptr::non_null::NonNull<[u8]>, alloc::collections::TryReserveError>", align 8 
          204:  %_3 = load i64, ptr %self, align 8, !noundef !3 
          205:  tail call void @llvm.experimental.noalias.scope.decl(metadata !11) 
          206:  %0 = tail call { i64, i1 } @llvm.uadd.with.overflow.i64(i64 %_3, i64 1) 
          207:  %_28.1.i = extractvalue { i64, i1 } %0, 1 
          208:  br i1 %_28.1.i, label %bb2, label %bb13.i 
          209:  
          210: bb13.i: ; preds = %start 
          211:  %_28.0.i = extractvalue { i64, i1 } %0, 0 
          212:  %_15.0.i = shl i64 %_3, 1 
          213:  %_0.0.sroa.speculated.i.i = tail call i64 @llvm.umax.i64(i64 %_15.0.i, i64 %_28.0.i) 
          214:  %_0.0.sroa.speculated.i16.i = tail call i64 @llvm.umax.i64(i64 %_0.0.sroa.speculated.i.i, i64 8) 
          215:  %_4.i.i = icmp sgt i64 %_0.0.sroa.speculated.i16.i, -1 
          216:  %_0.sroa.0.0.i.i = zext i1 %_4.i.i to i64 
          217:  call void @llvm.lifetime.start.p0(i64 24, ptr nonnull %self3.i), !noalias !11 
          218:  call void @llvm.lifetime.start.p0(i64 24, ptr nonnull %_20.i), !noalias !11 
          219:  tail call void @llvm.experimental.noalias.scope.decl(metadata !14) 
          220:  tail call void @llvm.experimental.noalias.scope.decl(metadata !17) 
          221:  %1 = icmp eq i64 %_3, 0 
          222:  br i1 %1, label %"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE.exit.i", label %bb4.i.i 
          223:  
          224: bb4.i.i: ; preds = %bb13.i 
          225:  %2 = getelementptr inbounds i8, ptr %self, i64 8 
          226:  %self2.i.i = load ptr, ptr %2, align 8, !alias.scope !19, !noalias !14, !nonnull !3, !noundef !3 
          227:  store ptr %self2.i.i, ptr %_20.i, align 8, !alias.scope !14, !noalias !19 
          228:  %_9.sroa.5.0._0.sroa_idx.i.i = getelementptr inbounds i8, ptr %_20.i, i64 16 
          229:  store i64 %_3, ptr %_9.sroa.5.0._0.sroa_idx.i.i, align 8, !alias.scope !14, !noalias !19 
          230:  br label %"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE.exit.i" 
          231:  
          232: "_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE.exit.i": ; preds = %bb4.i.i, %bb13.i 
          233:  %.sink.i.i = phi i64 [ 1, %bb4.i.i ], [ 0, %bb13.i ] 
          234:  %3 = getelementptr inbounds i8, ptr %_20.i, i64 8 
          235:  store i64 %.sink.i.i, ptr %3, align 8, !alias.scope !14, !noalias !19 
          236: ; call alloc::raw_vec::finish_grow 
          237:  call fastcc void @_ZN5alloc7raw_vec11finish_grow17h4a044e2f20ebcde0E(ptr noalias nocapture noundef nonnull sret([24 x i8]) align 8 dereferenceable(24) %self3.i, i64 noundef %_0.sroa.0.0.i.i, i64 %_0.0.sroa.speculated.i16.i, ptr noalias nocapture noundef nonnull readonly align 8 dereferenceable(24) %_20.i) 
          238:  call void @llvm.lifetime.end.p0(i64 24, ptr nonnull %_20.i), !noalias !11 
          239:  %_44.i = load i64, ptr %self3.i, align 8, !range !20, !noalias !11, !noundef !3 
          240:  %trunc.not.i = icmp eq i64 %_44.i, 0 
          241:  %4 = getelementptr inbounds i8, ptr %self3.i, i64 8 
          242:  br i1 %trunc.not.i, label %bb3, label %bb19.i 
          243:  
          244: bb19.i: ; preds = %"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE.exit.i" 
          245:  %e.0.i = load i64, ptr %4, align 8, !range !10, !noalias !11, !noundef !3 
          246:  %5 = getelementptr inbounds i8, ptr %self3.i, i64 16 
          247:  %e.1.i = load i64, ptr %5, align 8, !noalias !11 
          248:  call void @llvm.lifetime.end.p0(i64 24, ptr nonnull %self3.i), !noalias !11 
          249:  br label %bb2 
          250:  
          251: bb2: ; preds = %bb19.i, %start 
          252:  %_0.sroa.4.0.i.ph = phi i64 [ undef, %start ], [ %e.1.i, %bb19.i ] 
          253:  %_0.sroa.0.0.i.ph = phi i64 [ 0, %start ], [ %e.0.i, %bb19.i ] 
          254: ; call alloc::raw_vec::handle_error 
          255:  tail call void @_ZN5alloc7raw_vec12handle_error17h0fda3b0845d3b4baE(i64 noundef %_0.sroa.0.0.i.ph, i64 %_0.sroa.4.0.i.ph) #13 
          256:  unreachable 
          257:  
          258: bb3: ; preds = %"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE.exit.i" 
          259:  %v.0.i = load ptr, ptr %4, align 8, !noalias !11, !nonnull !3, !noundef !3 
          260:  call void @llvm.lifetime.end.p0(i64 24, ptr nonnull %self3.i), !noalias !11 
          261:  %6 = getelementptr inbounds i8, ptr %self, i64 8 
          262:  store ptr %v.0.i, ptr %6, align 8, !alias.scope !11 
          263:  store i64 %_0.0.sroa.speculated.i16.i, ptr %self, align 8, !alias.scope !11 
          264:  ret void 
          265: } 
          266:  
          267: ; Function Attrs: mustprogress nofree nosync nounwind nonlazybind willreturn memory(argmem: read, inaccessiblemem: readwrite) uwtable 
          268: define void @noop(ptr noalias nocapture noundef readonly align 8 dereferenceable(24) %v) unnamed_addr #4 personality ptr @rust_eh_personality { 
          269: start: 
          270:  %0 = getelementptr inbounds i8, ptr %v, i64 16 
          271:  %_2.i = load i64, ptr %0, align 8, !alias.scope !21, !noundef !3 
          272:  %.not = icmp eq i64 %_2.i, 0 
          273:  br i1 %.not, label %bb3, label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E.exit" 
          274:  
          275: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E.exit": ; preds = %start 
          276:  %_3.0.i = add i64 %_2.i, -1 
          277:  %1 = load i64, ptr %v, align 8, !noundef !3 
          278:  %cond.i = icmp ult i64 %_3.0.i, %1 
          279:  tail call void @llvm.assume(i1 %cond.i) 
          280:  br label %bb3 
          281:  
          282: bb3: ; preds = %start, %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E.exit" 
          283:  ret void 
          284: } 
          285:  
          286: ; Function Attrs: nonlazybind uwtable 
          287: define void @push_byte(ptr noalias nocapture noundef align 8 dereferenceable(24) %v) unnamed_addr #0 personality ptr @rust_eh_personality { 
check:22'0                            X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
check:22'0     ~~~~~~~
check:22'0     ~~~~~~~
          289:  tail call void @llvm.experimental.noalias.scope.decl(metadata !24) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          290:  %0 = getelementptr inbounds i8, ptr %v, i64 16 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          291:  %_4.i = load i64, ptr %0, align 8, !alias.scope !24, !noundef !3 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          292:  %1 = load i64, ptr %v, align 8, !alias.scope !24, !noundef !3 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          293:  %_3.i = icmp eq i64 %_4.i, %1 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          294:  br i1 %_3.i, label %bb1.i, label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E.exit" 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          296: bb1.i: ; preds = %start 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          297: ; call alloc::raw_vec::RawVec<T,A>::grow_one 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          298:  tail call fastcc void @"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$8grow_one17h1f9c44696f70c80fE"(ptr noalias noundef nonnull align 8 dereferenceable(16) %v) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          299:  %count.pre.i = load i64, ptr %0, align 8, !alias.scope !24 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          300:  br label %"_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E.exit" 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          302: "_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E.exit": ; preds = %start, %bb1.i 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          303:  %count.i = phi i64 [ %count.pre.i, %bb1.i ], [ %_4.i, %start ] 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          304:  %2 = getelementptr inbounds i8, ptr %v, i64 8 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          305:  %self1.i = load ptr, ptr %2, align 8, !alias.scope !24, !nonnull !3, !noundef !3 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          306:  %end.i = getelementptr inbounds i8, ptr %self1.i, i64 %count.i 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          307:  store i8 3, ptr %end.i, align 1, !noalias !24 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          308:  %_13.0.i = add i64 %count.i, 1 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          309:  store i64 %_13.0.i, ptr %0, align 8, !alias.scope !24 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          310:  ret void 
check:22'0     ~~~~~~~~~~
          311: } 
check:22'0     ~~
check:22'0     ~
check:22'0     ~
          313: ; Function Attrs: nonlazybind uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          314: declare noundef i32 @rust_eh_personality(i32 noundef, i32 noundef, i64 noundef, ptr noundef, ptr noundef) unnamed_addr #0 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
          316: ; core::fmt::num::imp::<impl core::fmt::Display for usize>::fmt 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          317: ; Function Attrs: nonlazybind uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          318: declare noundef zeroext i1 @"_ZN4core3fmt3num3imp54_$LT$impl$u20$core..fmt..Display$u20$for$u20$usize$GT$3fmt17hb88a4f59f684001dE"(ptr noalias noundef readonly align 8 dereferenceable(8), ptr noalias noundef align 8 dereferenceable(64)) unnamed_addr #0 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
          320: ; core::fmt::num::<impl core::fmt::UpperHex for usize>::fmt 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'1                               ?                                  possible intended match
          321: ; Function Attrs: nonlazybind uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          322: declare noundef zeroext i1 @"_ZN4core3fmt3num55_$LT$impl$u20$core..fmt..UpperHex$u20$for$u20$usize$GT$3fmt17hbf3ebfbcb0e69bc3E"(ptr noalias noundef readonly align 8 dereferenceable(8), ptr noalias noundef align 8 dereferenceable(64)) unnamed_addr #0 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
          324: ; core::fmt::num::<impl core::fmt::LowerHex for usize>::fmt 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          325: ; Function Attrs: nonlazybind uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          326: declare noundef zeroext i1 @"_ZN4core3fmt3num55_$LT$impl$u20$core..fmt..LowerHex$u20$for$u20$usize$GT$3fmt17hf43a2df9fec5b7c7E"(ptr noalias noundef readonly align 8 dereferenceable(8), ptr noalias noundef align 8 dereferenceable(64)) unnamed_addr #0 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
          328: ; core::panicking::panic_fmt 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          329: ; Function Attrs: cold noinline noreturn nonlazybind uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          330: declare void @_ZN4core9panicking9panic_fmt17h9a0b8708a3e6eb51E(ptr noalias nocapture noundef readonly align 8 dereferenceable(48), ptr noalias noundef readonly align 8 dereferenceable(24)) unnamed_addr #2 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          332: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          333: declare void @llvm.assume(i1 noundef) #5 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
          335: ; core::panicking::assert_failed_inner 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          336: ; Function Attrs: cold noinline noreturn nonlazybind uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          337: declare void @_ZN4core9panicking19assert_failed_inner17h157688ee9a381e38E(i8 noundef, ptr noundef nonnull align 1, ptr noalias noundef readonly align 8 dereferenceable(24), ptr noundef nonnull align 1, ptr noalias noundef readonly align 8 dereferenceable(24), ptr noalias nocapture noundef readonly align 8 dereferenceable(48), ptr noalias noundef readonly align 8 dereferenceable(24)) unnamed_addr #2 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          339: ; Function Attrs: mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          340: declare { i64, i1 } @llvm.uadd.with.overflow.i64(i64, i64) #6 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          342: ; Function Attrs: nounwind nonlazybind allockind("alloc,uninitialized,aligned") allocsize(0) uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          343: declare noalias noundef ptr @__rust_alloc(i64 noundef, i64 allocalign noundef) unnamed_addr #7 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          345: ; Function Attrs: nounwind nonlazybind allockind("realloc,aligned") allocsize(3) uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          346: declare noalias noundef ptr @__rust_realloc(ptr allocptr noundef, i64 noundef, i64 allocalign noundef, i64 noundef) unnamed_addr #8 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
          348: ; alloc::raw_vec::handle_error 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          349: ; Function Attrs: cold noreturn nonlazybind uwtable 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          350: declare void @_ZN5alloc7raw_vec12handle_error17h0fda3b0845d3b4baE(i64 noundef, i64) unnamed_addr #9 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          352: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          353: declare void @llvm.lifetime.start.p0(i64 immarg, ptr nocapture) #10 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          355: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          356: declare void @llvm.lifetime.end.p0(i64 immarg, ptr nocapture) #10 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          358: ; Function Attrs: nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          359: declare void @llvm.experimental.noalias.scope.decl(metadata) #11 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          361: ; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none) 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          362: declare i64 @llvm.umax.i64(i64, i64) #12 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          364: attributes #0 = { nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          365: attributes #1 = { inlinehint mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          366: attributes #2 = { cold noinline noreturn nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          367: attributes #3 = { noinline nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          368: attributes #4 = { mustprogress nofree nosync nounwind nonlazybind willreturn memory(argmem: read, inaccessiblemem: readwrite) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          369: attributes #5 = { mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          370: attributes #6 = { mustprogress nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          371: attributes #7 = { nounwind nonlazybind allockind("alloc,uninitialized,aligned") allocsize(0) uwtable "alloc-family"="__rust_alloc" "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          372: attributes #8 = { nounwind nonlazybind allockind("realloc,aligned") allocsize(3) uwtable "alloc-family"="__rust_alloc" "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          373: attributes #9 = { cold noreturn nonlazybind uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          374: attributes #10 = { mustprogress nocallback nofree nosync nounwind willreturn memory(argmem: readwrite) } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          375: attributes #11 = { nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: readwrite) } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          376: attributes #12 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          377: attributes #13 = { noreturn } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          378: attributes #14 = { nounwind } 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          380: !llvm.module.flags = !{!0, !1} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          381: !llvm.ident = !{!2} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~
check:22'0     ~
check:22'0     ~
          383: !0 = !{i32 8, !"PIC Level", i32 2} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          384: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          385: !2 = !{!"rustc version 1.79.0-nightly (ca8923a70 2024-03-29)"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          386: !3 = !{} 
check:22'0     ~~~~~~~~~
          387: !4 = !{i64 8} 
check:22'0     ~~~~~~~~~~~~~~
          388: !5 = !{!6} 
check:22'0     ~~~~~~~~~~~
          389: !6 = distinct !{!6, !7, !"_ZN4core3fmt3num52_$LT$impl$u20$core..fmt..Debug$u20$for$u20$usize$GT$3fmt17h2f7f534e69830b41E: %f"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          390: !7 = distinct !{!7, !"_ZN4core3fmt3num52_$LT$impl$u20$core..fmt..Debug$u20$for$u20$usize$GT$3fmt17h2f7f534e69830b41E"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          391: !8 = !{!9} 
check:22'0     ~~~~~~~~~~~
          392: !9 = distinct !{!9, !7, !"_ZN4core3fmt3num52_$LT$impl$u20$core..fmt..Debug$u20$for$u20$usize$GT$3fmt17h2f7f534e69830b41E: %self"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          393: !10 = !{i64 0, i64 -9223372036854775807} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          394: !11 = !{!12} 
check:22'0     ~~~~~~~~~~~~~
          395: !12 = distinct !{!12, !13, !"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14grow_amortized17he470d93f1fcfd8a3E: %self"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          396: !13 = distinct !{!13, !"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14grow_amortized17he470d93f1fcfd8a3E"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          397: !14 = !{!15} 
check:22'0     ~~~~~~~~~~~~~
          398: !15 = distinct !{!15, !16, !"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE: %_0"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          399: !16 = distinct !{!16, !"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          400: !17 = !{!18} 
check:22'0     ~~~~~~~~~~~~~
          401: !18 = distinct !{!18, !16, !"_ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$14current_memory17h8477196bbbe0428aE: %self"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          402: !19 = !{!18, !12} 
check:22'0     ~~~~~~~~~~~~~~~~~~
          403: !20 = !{i64 0, i64 2} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~
          404: !21 = !{!22} 
check:22'0     ~~~~~~~~~~~~~
          405: !22 = distinct !{!22, !23, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17h5b5b581158db68edE: %self"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          406: !23 = distinct !{!23, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$3pop17h5b5b581158db68edE"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          407: !24 = !{!25} 
check:22'0     ~~~~~~~~~~~~~
          408: !25 = distinct !{!25, !26, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E: %self"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          409: !26 = distinct !{!26, !"_ZN5alloc3vec16Vec$LT$T$C$A$GT$4push17hd890493dd35c4818E"} 
check:22'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

--- a/tests/codegen/vec_pop_push_noop.rs
+++ b/tests/codegen/vec_pop_push_noop.rs
@@ -5,10 +5,10 @@
 #[no_mangle]
 // CHECK-LABEL: @noop(
 pub fn noop(v: &mut Vec<u8>) {
-    // CHECK-NOT: reserve_for_push
+    // CHECK-NOT: grow_one
     // CHECK-NOT: call
     // CHECK: tail call void @llvm.assume
-    // CHECK-NOT: reserve_for_push
+    // CHECK-NOT: grow_one
     // CHECK-NOT: call
     // CHECK: ret
     if let Some(x) = v.pop() {
@@ -19,6 +19,6 @@ pub fn noop(v: &mut Vec<u8>) {
 #[no_mangle]
 // CHECK-LABEL: @push_byte(
 pub fn push_byte(v: &mut Vec<u8>) {
-    // CHECK: call {{.*}}reserve_for_push
+    // CHECK: call {{.*}}grow_one
     v.push(3);
 }

